File: joueur/Type.java
package joueur;


public enum Type {
    AUCUN("Aucun"),
    PHYSIQUE("Physique"),// 1
    LUMIERE("Lumiere"),// 1
    TENEBRE("Tenebre"),// 1

    EAU("Eau"), // 1
    FEU( "Feu" ), // 1
    PLANTE( "Plante" ), // 2

    QUANTIC("Quantique"), // 1
    ROCHE("Roche"); // 3
    
    private int id;
    private String type;
    

    Type(String type){
        this.type = type;
    }
    
    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }
    public String getType() {
        return type;
    }
    public void setType(String type) {
        this.type = type;
    }

   /* static{
        LUMIERE.faiblesse.addAll(Arrays.asList(Type.TENEBRE));

        TENEBRE.faiblesse.addAll(Arrays.asList(Type.LUMIERE));

        PHYSIQUE.faiblesse.addAll(Arrays.asList(Type.FEU, Type.ROCHE));

        FEU.faiblesse.addAll(Arrays.asList(Type.EAU ,Type.ROCHE));

        PLANTE.faiblesse.addAll(Arrays.asList(Type.FEU, Type.PHYSIQUE));

        EAU.faiblesse.addAll(Arrays.asList(Type.PLANTE));
    }*/

    

    

}

File: Cs/type/Type.cs
package type;

import java.util.HashSet;
import java.util.Set;

public enum Type {
    AUCUN("Aucun", 0),
    PHYSIQUE("Physique", 1),
    LUMIERE("Lumiere", 1),
    TENEBRE("Tenebre", 1),
    EAU("Eau", 1),
    FEU("Feu", 1),
    PLANTE("Plante", 2),
    QUANTIC("Quantique", 1),
    ROCHE("Roche", 3);

    private int id;
    private String type;
    private Set<Type> faiblesses;

    // Constructeur
    Type(String type, int id) {
        this.type = type;
        this.id = id;
        this.faiblesses = new HashSet<>();
    }

    public int getId() {
        return id;
    }

    public String getType() {
        return type;
    }

    public Set<Type> getFaiblesses() {
        return faiblesses;
    }

    public void addFaiblesse(Type faiblesse) {
        this.faiblesses.add(faiblesse);
    }

    // Initialisation des faiblesses
    static {
        LUMIERE.addFaiblesse(TENEBRE);
        TENEBRE.addFaiblesse(LUMIERE);

        PHYSIQUE.addFaiblesse(FEU);
        PHYSIQUE.addFaiblesse(ROCHE);

        FEU.addFaiblesse(EAU);
        FEU.addFaiblesse(ROCHE);

        PLANTE.addFaiblesse(FEU);
        PLANTE.addFaiblesse(PHYSIQUE);

        EAU.addFaiblesse(PLANTE);
    }
}
